src/catalog/src/local/memory.rs

```diff
@@ -264,7 +218,7 @@ impl MemoryCatalogManager {
     }
 
     /// Registers a schema and returns an error if the catalog or schema does not exist.
-    pub fn register_table(&self, request: RegisterTableRequest) -> Result<bool> {
+    pub fn register_table_sync(&self, request: RegisterTableRequest) -> Result<bool> {
```

Comment:
The `sync` postfix is unnecessary. And `sync` has a different meaning in Rust that may cause confusion.